packages/rpc-api/src/simulateTransaction.ts

```diff
@@ -95,7 +107,22 @@ type SimulateTransactionApiResponseWithAccounts<T extends AccountInfoBase> = Sol
     accounts: (T | null)[];
 }>;
 
+type SimulateTransactionApiResponseWithInnerInstructions = SolanaRpcResponse<
+    TransactionForFullMetaInnerInstructionsParsed | TransactionForFullMetaInnerInstructionsUnparsed
+>;
+
```

Comment:
Starting to regret what I might have signed up for with [this comment](https://github.com/solana-labs/solana-web3.js/pull/2865#discussion_r1657944926)....